package/MDAnalysis/core/universe.py

```diff
@@ -363,15 +374,32 @@ def __init__(self, topology=None, *coordinates, all_coordinates=False,
 
         if coordinates:
             self.load_new(coordinates, format=format, in_memory=in_memory,
-                        in_memory_step=in_memory_step, **kwargs)
+                          in_memory_step=in_memory_step, **kwargs)
 
         if transformations:
             if callable(transformations):
                 transformations = [transformations]
             self._trajectory.add_transformations(*transformations)
 
-        if guess_bonds:
-            self.atoms.guess_bonds(vdwradii=vdwradii)
+        to_guess = list(to_guess)
+        self._topology_atrrs = list(
+            att.attrname for att in self._topology.read_attributes)
+
+        if ('MINIMAL' not in self._parser and 'THINGY' not in self._parser and
+            self._topology.n_atoms > 0 and any(fmt in _PARSERS for fmt in self._parser)):
+
+            
+            if (not any(att == 'types' for att in self._topology_atrrs) and
+                    'types' not in to_guess):
+                to_guess.append('types')
+            if (not any(att == 'masses' for att in self._topology_atrrs) and
+                    'masses' not in to_guess):
+                to_guess.append('masses')
```

Comment:
I agree with adding types and masses in the to_guess parameter and removing the checking part regarding them.
However, If we removed the parsers checkpoint, the problem that will arise here is that whenever a universe is created (empty universe, trashy input, etc.), the guess_TopologyAttributes will be automatically called to guess masses and types. I was trying to prevent this behavior from happening to avoid unnecessary errors, but now I think these errors should be handled inside the API itself, so I'll try to add it inside the `guess_topologyAttribute` API